cypress/integration/tests/administrator/repository/git.test.ts

```diff
@@ -0,0 +1,126 @@
+import * as data from "../../../../utils/data_utils";
+import { GitConfiguration } from "../../../models/administrator/repositories/git";
+import { Analysis } from "../../../models/developer/applicationinventory/analysis";
+import { CredentialsSourceControlUsername } from "../../../models/administrator/credentials/credentialsSourceControlUsername";
+import { CredentialType, UserCredentials } from "../../../types/constants";
+
+let gitConfiguration = new GitConfiguration();
+let source_credential;
+
+describe("Test an application form a Git source", { tags: "@tier1" }, () => {
+    before("Login", function () {
+        // Prevent hook from running, if the tag is excluded from run
+        if (hasToBeSkipped("@tier1")) return;
+
+        login();
+        deleteApplicationTableRows();
+        source_credential = new CredentialsSourceControlUsername(
+            data.getRandomCredentialsData(
+                CredentialType.sourceControl,
+                UserCredentials.usernamePassword,
+                true
+            )
+        );
+        source_credential.create();
+    });
+
+    beforeEach("Persist session", function () {
+        // Save the session and token cookie for maintaining one login session
+        preservecookies();
+        cy.fixture("application").then(function (appData) {
+            this.appData = appData;
+        });
+        cy.fixture("analysis").then(function (analysisData) {
+            this.analysisData = analysisData;
+        });
+
+        // Interceptors
+        cy.intercept("POST", "/hub/application*").as("postApplication");
+        cy.intercept("GET", "/hub/application*").as("getApplication");
+    });
+
+    afterEach("Persist session", function () {
+        // Reset URL from report page to web UI
+        resetURL();
+    });
+
+    after("Perform test data clean up", () => {
+        if (hasToBeSkipped("@tier1")) return;
+        deleteApplicationTableRows();
+        deleteAllBusinessServices();
+        source_credential.delete();
+        writeMavenSettingsFile(data.getRandomWord(5), data.getRandomWord(5));
+    });
+
+    it("Enable Insecure git Repository", () => {
+        GitConfiguration.open();
+        gitConfiguration.toggleInsecureGitRepositories();
+    });
+
+    it("Source code analysis on tackle testapp", function () {
+        // For tackle test app source credentials are required.
+        const application = new Analysis(
+            getRandomApplicationData("Secure_enabled_tackle_test_app", {
+                sourceData: this.appData[3],
+            }),
+            getRandomAnalysisData(this.analysisData[0])
+        );
+        application.create();
+        cy.wait("@getApplication");
+        cy.wait(2000);
+        application.manageCredentials(source_credential.name, "None");
+        application.analyze();
+        application.verifyAnalysisStatus("Completed");
+        application.openreport();
+    });
+
+    it("Disable insecure git Repository", () => {
```

Comment:
it("Analysis on insecure git Repository(http) for tackle test app",

and then combine with below test case